alua.c

```diff
@@ -560,17 +561,55 @@ int alua_implicit_transition(struct tcmu_device *dev, struct tcmulib_cmd *cmd)
 		return ret;
 
 	pthread_mutex_lock(&rdev->state_lock);
-	if (rdev->lock_state == TCMUR_DEV_LOCK_LOCKED) {
+	if (rdev->lock_state == TCMUR_DEV_LOCK_WRITE_LOCKED) {
+		/* For both read/write cases in this state is good */
 		goto done;
-	} else if (rdev->lock_state == TCMUR_DEV_LOCK_LOCKING) {
-		tcmu_dev_dbg(dev, "Lock acquisition operation is already in process.\n");
+	} else if (rdev->lock_state == TCMUR_DEV_LOCK_WRITE_LOCKING) {
+		/* For both read/write cases in this state should return busy */
+		tcmu_dev_dbg(dev, "Write lock acquisition operation is already in process.\n");
 		ret = TCMU_STS_BUSY;
 		goto done;
+	} else if (is_read && !rdev->lock_lost) {
```

Comment:
Nit: does this `lock_lost` check matter? It looks like `__tcmu_notify_lock_lost` will set this to `true` and set the state back to `UNLOCKED`.

Reply:
Hmm, I will remove this.